_1327/documents/utils.py

```diff
@@ -41,26 +39,24 @@ def handle_edit(request, document):
 
 			return True, form
 	else:
-
 		# load Autosave
+		autosave = None
 		try:
 			autosave = TemporaryDocumentText.objects.get(document=document)
-			text = autosave.text
 			autosaved = True
 		except TemporaryDocumentText.DoesNotExist:
-			text = document.text
 			autosaved = False
 
-		if 'restore' not in request.GET:
-			text = document.text
-		else:
+		if 'restore' in request.GET:
 			autosaved = False
 
-		form_data = {
-			'title': document.title,
-			'text': text,
-		}
-		form = TextForm(form_data)
+		if 'restore' in request.GET and autosave is not None:
+			form_data = {
+				'text': autosave.text,

```

Comment:
It's now fixed in line 57. (The use of the ``initial`` dictionary overwrites the text, all other fields are still taken from the ``document`` model)